src/modules/endpoint/reducers/cityContentReducer.js

```diff
@@ -27,61 +27,84 @@ export default (
         throw Error('Cannot push category on not initialized cityContent')
       }
       return pushCategory(state, action)
+    case 'PUSH_CATEGORY_LANGUAGES': {
+      if (state === null) {
+        throw Error('Cannot push category languages on not initialized cityContent')
+      }
+      const { key, ...rest } = action.params
+      return {
+        ...state,
+        categoriesRouteMapping: {
+          ...state.categoriesRouteMapping,
+          [key]: { ...rest, status: 'languageNotAvailable' }
+        }
+      }
+    }
     case 'PUSH_EVENT':
       if (state === null) {
         throw Error('Cannot push event on not initialized cityContent')
       }
       return pushEvent(state, action)
+    case 'PUSH_EVENT_LANGUAGES':
+      if (state === null) {
+        throw Error('Cannot push event languages on not initialized cityContent')
+      }
+      const { key, ...rest } = action.params
+      return {
+        ...state,
+        eventsRouteMapping: {
+          ...state.eventsRouteMapping,
+          [key]: { ...rest, status: 'languageNotAvailable' }
+        }
+      }
     case 'MORPH_CONTENT_LANGUAGE':
       if (state === null) {
         throw Error('Cannot morph content language on not initialized cityContent')
       }
       return morphContentLanguage(state, action)
     case 'FETCH_EVENT': {
       const { language, path, key, city } = action.params
-      const newState = state === null ? createCityContent(city) : state
+      const newState = state === null ? createCityContent(city) : { ...state }
       newState.eventsRouteMapping[key] = { status: 'loading', language, city, path }
       return newState
     }
     case 'CLEAR_EVENT': {
       const { key } = action.params
       if (state === null) {
-        return state
+        return null
       }
       delete state.eventsRouteMapping[key]
-      return state
+      return { ...state }
```

Comment:
Why did you change this?

Reply:
Reducers mustn't mutate the parameters (see https://redux.js.org/basics/reducers#handling-actions).
I modified the types now, such that flow prevents stuff like that now.